Jenkinsfile

```diff
@@ -43,458 +49,599 @@ pipeline {
           if (GIT_LOCAL_BRANCH != "develop") {
             def builds = load ".jenkinsci/cancel-builds-same-job.groovy"
             builds.cancelSameJobBuilds()
           }
         }
       }
     }
-    stage('Build Debug') {
-      when {
-        allOf {
-          expression { params.build_type == 'Debug' }
-          expression { return params.iroha }
-        }
-      }
+    stage('Build') {
       parallel {
-        stage ('x86_64_linux') {
+        stage ('Linux') {
           when {
             beforeAgent true
-            expression { return params.x86_64_linux }
+            anyOf {
+              expression { return params.Linux }
+              allOf {
+                expression { return env.CHANGE_ID != null }
+                expression { return env.GIT_PREVIOUS_COMMIT != null }
+                expression { return env.CHANGE_TARGET ==~ /(master|develop|trunk)/ }
+                expression { return params.Merge_PR }
+              }
+            }
           }
-          agent { label 'x86_64' }
+          agent { label 'x86_64_aws_build' }
           steps {
             script {
-              debugBuild = load ".jenkinsci/debug-build.groovy"
-              coverage = load ".jenkinsci/selected-branches-coverage.groovy"
-              if (coverage.selectedBranchesCoverage(['develop', 'master'])) {
-                debugBuild.doDebugBuild(true)
-              }
-              else {
-                debugBuild.doDebugBuild()
+              def debugBuild = load ".jenkinsci/debug-build.groovy"
+              def coverage = load ".jenkinsci/selected-branches-coverage.groovy"
+              if (params.BUILD_TYPE == 'Debug') {
+                if (params.Merge_PR && env.CHANGE_TARGET ==~ /master/) {
+                  debugBuild.doDebugBuild(true)
+                }
+                else {
+                  debugBuild.doDebugBuild(coverage.selectedBranchesCoverage())
+                }
               }
-              if (GIT_LOCAL_BRANCH ==~ /(master|develop)/) {
-                releaseBuild = load ".jenkinsci/release-build.groovy"
+              if (params.BUILD_TYPE == 'Release') {
+                def releaseBuild = load ".jenkinsci/release-build.groovy"
                 releaseBuild.doReleaseBuild()
               }
             }
           }
           post {
-            always {
+            success {
               script {
-                post = load ".jenkinsci/linux-post-step.groovy"
-                post.linuxPostStep()
+                if (params.BUILD_TYPE == 'Release') {
+                  def post = load ".jenkinsci/linux-post-step.groovy"
+                  post.postStep()
+                }
               }
             }
           }
         }
-        stage('armv7_linux') {
+        stage('ARMv7') {
           when {
             beforeAgent true
-            expression { return params.armv7_linux }
+            anyOf {
+              expression { return params.ARMv7 }
+              // allOf {
```

Comment:
Please add a TODO to fix issues and uncomment, with Jira task id. Same for ARMv8 below.